File: selection_sort/selection_sort.py
#!/usr/bin/python3

from typing import List


def selection_sort(nums: List[int]) -> List[int]:
    out = []

    while nums:
        lowest = nums[0]
        lowest_idx = 0

        for i in range(0, len(nums)):
            if nums[i] < lowest:
                lowest = nums[i]
                lowest_idx = i

        out.append(nums.pop(lowest_idx))

    return out

File: selection_sort/selection_sort.go
package selection_sort

func selectionSort(in []int) []int {
	out := make([]int, len(in))
	outIndex := 0

	for len(in) > 0 {
		low := in[0]
		lowIndex := 0

		for i := range in {
			if in[i] < low {
				low = in[i]
				lowIndex = i
			}
		}

		out[outIndex] = low
		outIndex++

		in = append(in[:lowIndex], in[lowIndex+1:]...)
	}

	return out
}
